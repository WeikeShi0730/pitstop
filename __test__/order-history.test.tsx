import { render, RenderResult } from "@testing-library/react";
import OrderHistory from "../components/order-history.component";

let documentBody: RenderResult;
describe("<OrderHistory />", () => {
  beforeEach(
    () =>
      (documentBody = render(
        <OrderHistory
          orderHistoryItems={[
            {
              items: [
                {
                  product: {
                    imageUrl: "test",
                    name: "",
                    price: 0,
                    id: "",
                    featured: false,
                    sold: 0,
                    teamId: "",
                  },
                  count: 10,
                },
                {
                  product: {
                    imageUrl: "test",
                    name: "",
                    price: 0,
                    id: "",
                    featured: false,
                    sold: 0,
                    teamId: "",
                  },
                  count: 5,
                },
              ],
              timeStamp: "2022-02-22",
            },
          ]}
        />
      ))
  );
  it("Test <OrderHistory />", () => {
    expect(documentBody.queryByText("2022-02-22")).toBeInTheDocument();
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
