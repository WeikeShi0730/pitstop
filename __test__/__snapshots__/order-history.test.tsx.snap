// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderHistory /> Test <OrderHistory /> 1`] = `
<body>
  <div>
    <div
      class="flex flex-col h-auto lg:h-full w-full"
    >
      <div
        class="h-auto grid grid-cols-1 gap-10 h-full justify-items-center m-10"
      >
        <p
          class="flex justify-self-start text-slate-700 text-lg md:text-2xl col-span-1 py-1 border-b border-slate-700"
        >
          My order history
        </p>
        <div
          class="w-full max-w-xs sm:max-w-sm md:max-w-md lg:max-w-lg"
        >
          <button
            aria-expanded="false"
            class="font-light flex justify-between items-center w-full px-4 py-2 text-left text-slate-700 bg-slate-400 rounded-lg bg-opacity-50 backdrop-blur-sm hover:bg-slate-400 transition-all ease-in-out duration-200"
            id="headlessui-disclosure-button-1"
            type="button"
          >
            <span
              class="text-base md:text-xl"
            >
              2022-02-22
            </span>
            <svg
              class=" w-5 h-5"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 10.828l-4.95 4.95-1.414-1.414L12 8l6.364 6.364-1.414 1.414z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
      <div
        class="flex justify-center m-10 mt-auto"
      >
        <div>
          <div
            aria-orientation="horizontal"
            class="flex w-fit max-w-full p-1 gap-x-1 bg-slate-300 rounded-lg shadow text-sm md:text-base"
            role="tablist"
          >
            <button
              class="buttonPrev w-8 py-1 font-light leading-5 text-slate-700 rounded-lg hover:bg-slate-100 text-slate-400 hover:bg-transparent"
              disabled=""
            >
              &lt;
            </button>
            <button
              aria-selected="true"
              class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 bg-orange-theme shadow text-slate-100"
              id="headlessui-tabs-tab-3"
              role="tab"
              tabindex="0"
              type="button"
            >
              1
            </button>
            <button
              class="buttonNext w-8 py-1 font-light leading-5 text-slate-700 rounded-lg hover:bg-slate-100 text-slate-400 hover:bg-transparent"
              disabled=""
            >
              &gt;
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
